src/widgets/mfa/MfaCredentialsWidget.jsx

```diff
@@ -24,42 +25,83 @@ const EmailVerificationCodeForm = createForm({
     ],
 })
 
+const PhoneNumberRegisteringCredentialForm = config => createForm({
+    prefix: 'r5-mfa-credentials-phone-number-',
+    fields: [
+        phoneNumberField({required: true}, config)
+    ],
+    submitLabel: 'mfa.register.phoneNumber'
+})
 class MainView extends React.Component {
     onEmailRegistering = _ => {
         return this.props.apiClient.startMfaEmailRegistration({
                 accessToken: this.props.accessToken
             }
         )
     }
+
+    onPhoneNumberRegistering = data => {
+        return this.props.apiClient.startMfaPhoneNumberRegistration({
+            accessToken: this.props.accessToken,
+            ...data
+        })
+    }
+
     render() {
         const { i18n, showIntro, config } = this.props
+        const PhoneNumberInputForm = PhoneNumberRegisteringCredentialForm(config);
         return (
             <div>
-                {config.mfaEmailEnabled && showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
-                {config.mfaEmailEnabled && <EmailRegisteringCredentialForm handler={this.onEmailRegistering} onSuccess={data => this.props.goTo('verification-code', data)}/>}
+                <div>
+                    {config.mfaEmailEnabled && showIntro && <Intro>{i18n('mfa.email.explain')}</Intro>}
```

Comment:
Does this require a change in SDK core? I believe settings are retrieved with some SDK core function whose model probably doesn't include `mfaEmailEnabled` yet.

Reply:
The config includes the new field without requiring any modification to the sdk core.